netbox/users/utils.py

```diff
@@ -0,0 +1,9 @@
+from social_core.storage import NO_ASCII_REGEX, NO_SPECIAL_REGEX
+
+
+def clean_username(value):
+    """Clean username removing any unsupported character"""
+    value = NO_ASCII_REGEX.sub('', value)
+    value = NO_SPECIAL_REGEX.sub('', value)
```

Comment:
We still want to allow `@` and `.` per Django's [`UnicodeUsernameValidator`](https://github.com/django/django/blob/a142edcc4981bb27f4356acad0c7fdfb2d5f9849/django/contrib/auth/validators.py#L19) validator. We can probably just reference its `regex` attribute to strip out any non-confirming characters.